module.js

```diff
@@ -2,12 +2,64 @@
 
 module.exports = {
 
+  extends: ['plugin:import/errors'],
+
+  plugins: ['import'],
+
   parserOptions: {
     sourceType: 'module'
   },
 
   rules: {
-    'func-name-matching': [2, {includeCommonJSModuleExports: true}]
-  }
+    'func-name-matching': [2, {includeCommonJSModuleExports: true}],
+
+    // Errors
+    'import/no-absolute-path': [2],
+    'import/no-cycle': [2],
+    'import/no-duplicates': [2],
+    'import/no-self-import': [2],
+
+    // Best practices
+    'import/export': [2],
+    'import/no-amd': [2],
+    'import/no-commonjs': [2],
+    'import/no-extraneous-dependencies': [2],
+    'import/no-import-module-exports': [2],
+    'import/no-mutable-exports': [2],
+    'import/no-named-as-default': [2],
+    'import/no-unused-modules': [2],
+    'import/no-useless-path-segments': [2],
+    'import/no-webpack-loader-syntax': [2],
+
+    // Style guide
+    'import/extensions': [2, 'ignorePackages'],
+    'import/first': [2],
+    'import/newline-after-import': [2],
+    'import/no-default-export': [2],
+    'import/no-unassigned-import': [2],
+    'import/order': [2, {
+      groups: ['builtin', 'external', 'parent', 'sibling', 'index', 'object'],
+      alphabetize: {order: 'asc'},
+    }],
```

Comment:
The import order here is somewhat opinionated, but I think it doesn't hurt anyone to have consistent import order, and this rule is auto-fixable. One thing we might consider is to add `space-between: always | never` so that the auto-fixer will also handle newlines correctly.